superset-frontend/src/chart/chartAction.js

```diff
@@ -395,7 +395,16 @@ export function exploreJSON(
         if (isFeatureEnabled(FeatureFlag.GLOBAL_ASYNC_QUERIES)) {
           // deal with getChartDataRequest transforming the response data
           const result = 'result' in response ? response.result[0] : response;
-          return waitForAsyncData(result);
+          const awaitingChartData =
+            'job_id' in result && 'channel_id' in result;
```

Comment:
Can we switch on HTTP status code instead here?

Reply:
Yep, good suggestion!